meerk40t/gui/simulation.py

```diff
@@ -383,14 +419,54 @@ def fit_scene_to_panel(self):
         )
         self.widget_scene.request_refresh()
 
+    def progress_to_idx(self, progress):
+        residual = 0
+        idx = progress
+        if not self._playback_cuts:
+            # progress is the time indicator
+            idx = len(self.statistics) - 1
+            prev_time = None
+            while idx >= 0:
+                item = self.statistics[idx]
+                this_time = item["time_at_end_of_burn"]
+                # print (f"{idx} {this_time} vs {progress} - {item}")
+                if this_time <= progress:
+                    if prev_time is not None:
+                        # We compute a 0 to 1 ratio of the progress
+                        residual = (progress - this_time) / (prev_time - this_time)
+                    break
+                prev_time = this_time
+                idx -= 1
+            idx += 1
+            if idx == 0:
+                item = self.statistics[idx]
+                start_time = item["time_at_start"]
+                this_time = item["time_at_end_of_burn"]
+                residual = (progress - start_time) / (this_time - start_time)
+
+        if idx >= len(self.statistics):
+            idx = len(self.statistics) - 1
+        if idx < 0:
+            idx = 0
+        # print(
+        #     f"Cut-Mode={self._playback_cuts}, prog={progress}, idx={idx}, stats={len(self.statistics)}"
+        # )
+        return idx, residual
+
     def on_mouse_right_down(self, event=None):
+        def cut_before(event):
+            step, residual = self.progress_to_idx(self.progress)
+            self.context(f"plan{self.plan_name} sublist {step} -1\n")
```

Comment:
Does this make sense in time mode? If we are not at a specific point in the cutcode where we can delete half the cutcode and be at this location, we might well try to do a `delete cuts before` partway through a raster or something which could be a bit strange. Though disabling this feature in that mode would suffice.